test/src/lib/update-lock.spec.ts

```diff
@@ -242,6 +242,45 @@ describe('lib/update-lock', () => {
 			);
 		});
 
+		it('locks all applications before resolving input function', async () => {
+			const appIds = [111, 222, 333];
+			// Set up fake filesystem for lockfiles
+			mockFs({
+				[path.join(
+					constants.rootMountPoint,
+					updateLock.lockPath(111),
+					serviceName,
+				)]: {},
+				[path.join(
+					constants.rootMountPoint,
+					updateLock.lockPath(222),
+					serviceName,
+				)]: {},
+				[path.join(
+					constants.rootMountPoint,
+					updateLock.lockPath(333),
+					serviceName,
+				)]: {},
+			});
+
+			await expect(
+				updateLock.lock(appIds, { force: false }, async () => {
+					// At this point the locks should be taken and not removed
+					// until this function has been resolved
+					// Both `updates.lock` and `resin-updates.lock` should have been taken
+					expect(lockSpy.args).to.have.length(6);
```

Comment:
other tests use the `expectLocks` function which I wanted to but for some reason when it reads the folders from mockfs the locks do not exist. This test logic is checking the same thing without requiring any mocks/stubs to be working (maybe something is wrong between lock, lockfile, and mockfs.

@cywang117

Reply:
@20k-ultra I think expectLocks only checks for the single `appId` const declared in this test suite, the `appId` should be 1234567, so it makes sense why it wouldn't work for other appIds. The way this test is formatted works as-is, or you could also modify `expectLocks` -- up to you.

Reply:
yeah I modified it to check multiple applications Ids. It was 100% checking fs.readdir for each service but for some reason was saying the other applications were empty (only the first service checked would have locks). It's as is mockFS could only mock 1 folder at a time...

Reply:
MockFS can definitely mock multiple, just like you declared it [here](https://github.com/balena-os/balena-supervisor/pull/1953/files#diff-3e38a1b1da5c2f1d2fbe196b5b98996a7f6b8fa9721fb8de1c0f6f9da8b14aceR248-R264).. probably the way I implemented `mockLockDir` can only mock 1 initial directory. I think if you call mockFs repeatedly without mockFs.restore(), which may have been what you tried, the subsequent mock calls won't do anything. Instead, to add other directories to the mock after mockFs has already been called, you may have to use fs.mkdir or other fs methods..

Reply:
Anyways, since this is the only test suite that mocks multiple lock dirs, it's probably not necessary to modify `mockLockDir`, and the implementation you have here works fine 👍🏼